test: Bridge e2e automated tests

davibroc opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/29769?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

davibroc committed:
Refactored the tests

davibroc committed:
More changes

davibroc committed:
Finish refactor of the tests

davibroc committed:
Added mocking for second transaction

racitores committed:
ci: add new optional bridge e2e

racitores committed:
ci: test ci

racitores committed:
ci: test ci

racitores committed:
ci: test ci

davibroc committed:
Fixed import paths and changed Tenderly RPC

racitores committed:
ci: revert to all jobs

davibroc committed:
Remove @no-mmi

davibroc committed:
Increased the timeout for ETH funds to be available

davibroc committed:
Increase time out for waiting the transaction to complete

davibroc committed:
Refactored waiting for transaction code

davibroc committed:
Fixed Lint errors

davibroc committed:
Fixed Unit tests

davibroc committed:
Fixed more Unit tests

davibroc committed:
Fixed more lint errors

davibroc committed:
Add assert declaration

micaelae commented on ui/pages/bridge/prepare/bridge-input-group.tsx:
```diff
@@ -199,6 +200,7 @@ export const BridgeInputGroup = ({
                 networkImageSrc={networkImageSrc}
                 asset={(token as never) ?? undefined}
                 networkProps={networkProps}
+                data-testid="bridge-source-button"
```
When a dest token is selected this will be rendered. What do you think of passing in a `dataTestId` prop from prepare-bridge page instead? That value can then be used here and on line 187

davibroc replied:
Dond @micaelae please have a look

davibroc committed:
Changed data-testId setting

davibroc committed:
Fix lint

davibroc committed:
Fixed merge conflicts

davibroc committed:
Change tests to run on Anvil (#31173)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31173?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

davibroc committed:
Removed from optional run

davibroc committed:
Fixes

davibroc committed:
Added contract back

davibroc committed:
Added mock for getTokens

davibroc committed:
Changed bridge to Arbitrum One

davibroc committed:
Changed Bridge to Arbitrum One

davibroc committed:
Added getQuote mock

davibroc committed:
Fixed getQuote mock

davibroc committed:
Fix lint:error

davibroc committed:
Fixed lint errors

racitores reviewed (approved):
LGTM. Amazing job!
